flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/SimpleConsumerThread.java

```diff
@@ -460,15 +433,13 @@ public void cancel() {
 	// ------------------------------------------------------------------------
 
 	/**
-	 * Request latest offsets for a set of partitions, via a Kafka consumer.
-	 *
-	 * <p>This method retries three times if the response has an error.
+	 * Request offsets before a specific time for a set of partitions, via a Kafka consumer.
 	 *
 	 * @param consumer The consumer connected to lead broker
 	 * @param partitions The list of partitions we need offsets for
 	 * @param whichTime The type of time we are requesting. -1 and -2 are special constants (See OffsetRequest)
 	 */
-	private static void getLastOffsetFromKafka(
+	private static void requestAndSetSpecificTimeOffsetsFromKafka(
```

Comment:
Refactored the utility Kafka request methods in this class to avoid creating redundant lists.